src/resolve/local.ts

```diff
@@ -17,61 +13,35 @@ export default async function resolveLocal (spec: PackageSpec, opts: ResolveOpti
 
   if (dependencyPath.slice(-4) === '.tgz' || dependencyPath.slice(-7) === '.tar.gz') {
     const name = getTarballName(dependencyPath)
-    return {
+    const resolution: Resolution = {
+      type: 'tarball',
       id: createLocalPkgId(name, dependencyPath),
-      root: path.dirname(dependencyPath),
-      fetch: createLocalTarballFetcher({
-        tarball: dependencyPath
-      })
+      tarball: `file:${dependencyPath}`,
     }
+    return {resolution}
   }
 
   if (opts.linkLocal) {
```

Comment:
In that case we won't preserve the info in a shrinkwrap if the package should be linked or not. The question is if it's useful? I can see it being useful in a monorepo — you setup a workflow, run few `pnpm link` and then commit shrinkwrap — on a fresh clone you'll get everything linked for you automatically.